server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -581,4 +603,153 @@ public Function<String, String> lookupNameBySourceKey() {
             return Function.identity();
         }
     }
+
+
+    public DocTableInfo dropColumns(Collection<DropColumn> columns) {
+        for (var column : columns) {
+            ColumnIdent colToDrop = column.ref().column();
+            for (var generatedColumn : generatedColumns) {
+                if (Symbols.containsColumn(generatedColumn.generatedExpression(), colToDrop)) {
+                    throw new UnsupportedOperationException(String.format(
+                        Locale.ENGLISH,
+                        "Cannot drop column `%s`. It's used in generated column `%s`: %s",
+                        colToDrop,
+                        generatedColumn.ident().columnIdent().fqn(),
+                        generatedColumn.formattedGeneratedExpression()
+                    ));
```

Comment:
This looks like replacement for `AlterTableDropColumnAnalyzer.validateDynamic `call.
If not possible to re-use the whole method, can we extract this check into re-usable method?

Also, looks like "is part of INDEX" check (which is part of `validateDynamic`) is missing now?

Reply:
I integrated `validateDynamic` into the `DocTableInfo.dropColumns` and simply call that now from the analyzer as well.